### ISSUE TITLE
image formats

### ISSUE THREAD

**Issue by stephenmoloney** [kubenow/KubeNow#357]
### Issue
Some cloud providers do not accept `qcow2` image format

### Potential solution(s)

- release multiple image formats and make downloadable format configurable

- let the client download the `.qcow2` image format and do any conversions necessary prior to upload to the cloud provider

relates to #356

**Comment by mcapuccini** [comment]
Hi @stephenmoloney! Thanks for testing KubeNow, and for opening this issue. I assume we are talking about OpenStack here. 

Releasing `raw` images would be a bit harder IMO. First, it's more expensive, and second it's slower to save on the backend storage. Since we are using the free version  of Travis, builds need to be faster or they will be killed. Therefore, I'd be more keen to solution 2. 

I know that @andersla and @sneumann had this problem with image format in the past and they have found a workaround with it. Maybe it's time to merge efforts and get the logic officially supported.

**Comment by sneumann** [comment]
Hi, I didn't have a proper solution, and manually converted the image following 
https://docs.openstack.org/image-guide/convert-images.html
and manually uploaded to the images available. Yours, Steffen

**Comment by stephenmoloney** [comment]
@mcapuccini I'm not entirely satisfied that manual intervention is a good thing... As I suppose `KubeNow` is aiming to be `it just works` solution. If the client does the conversion, then perhaps it should be supported in some sort of script in the code too to do the conversion prior to upload.

@sneumann Do you know what size that upload was?

**Comment by mcapuccini** [comment]
@stephenmoloney agreed, it should be automatic. Pinging @andersla for any thoughts on this.

**Comment by andersla** [comment]
Yes I think it would be possible to quite easily download the qcow2 image into the user temp dir (as is don now already) and there convert it into raw and then continue with then upload.

**Comment by mcapuccini** [comment]
@stephenmoloney if you have slow connection, you could start an instance in OpenStack so you can do the first deployment from the tenancy. In this way download/upload of the image should be way faster. I like the idea of converting the image prior upload.

**Comment by stephenmoloney** [comment]
@mcapuccini That's sounds like a good idea, effectively, I take it you mean run `kn apply` inside a running instance in the opencloud stack. Beats waiting hours/days for an upload. 

So in that case, I'll make some changes to do a local conversion where the `disc_format` raw is specified in the `terraform variables` file. 

It might be a few days before I get a chance. I'll ping back when the PR is looking a bit more ready.

**Comment by mcapuccini** [comment]
Sure, no worries! Just let us know when it's ready 🙂

**Comment by kunalrdeshmukh** [comment]
Same error. Could you please let me know if you have fixed it? or else, how can i convert file to iso? i cant see it under tmp even after the download.

**Comment by mcapuccini** [comment]
I think @andersla has a solution for this. You could also try this: https://docs.openstack.org/image-guide/convert-images.html

**Comment by kunalrdeshmukh** [comment]
Thanks. So trick was to add `qemu-img convert -f qcow2 -O <desired_file_format> $filename <output file name>` command on line 70 of `bin/image-create-openstack.sh` file. 

bin folder is inside my_deployment folder which is created as a result of `kn init openstack my_deployment` command.
